guardian/newsletters-nx
# Use environment variables to control deployment of API

## What does this change?

Uses the following environment variables for the deployment of the API:
- NEWSLETTERS_UI_SERVE - checks for variable = true, or undefined and not in production
- NEWSLETTERS_API_READ_WRITE - checks for variable = true, or undefined and not in production
- NEWSLETTERS_API_READ - checks for variable = true, or undefined and not in production, or READ_WRITE variable = true, or undefined and not in production

Proxy now does not change path, which we specify directly in routes

## How to test

Run `npm install`
For each environment variable, change the value on the command line and invoke `npm run dev` e.g.
`NEWSLETTERS_UI_SERVE=false npm run dev`
Check that there is no access to the relevant options with the UI

## Have we considered potential risks?

Note that there is a problem when selecting the `api test page` option, but this problem already exists in the main branch and therefore will be fixed separately

## Images

` NEWSLETTERS_API_READ_WRITE=false npm run dev`
The wizard  is not displayed:

![Screenshot 2023-02-28 at 15 22 16](https://user-images.githubusercontent.com/74301289/221898218-9c3129b7-37cb-40a6-a551-2ba0305079b4.png)

` NEWSLETTERS_API_READ=false NEWSLETTERS_API_READ_WRITE=false npm run dev`
The list of current newsletters is not displayed:

![Screenshot 2023-02-28 at 15 24 14](https://user-images.githubusercontent.com/74301289/221898838-2a2420aa-59fd-4836-a1ca-ffa872b6f74d.png)
